Apps/W1/Shopify/app/src/Invoicing/Codeunits/ShpfyDraftOrdersAPI.Codeunit.al

```diff
@@ -0,0 +1,386 @@
+    /// <param name="ShpfyOrderTaxLines">Tax lines for a shopify order.</param>
+    /// <returns>Unique id of the created draft order in shopify.</returns>
+    internal procedure CreateDraftOrder(
+        var TempShpfyOrderHeader: Record "Shpfy Order Header" temporary;
+        var TempShpfyOrderLine: Record "Shpfy Order Line" temporary;
+        var ShpfyOrderTaxLines: Dictionary of [Text, Decimal]
+    ): BigInteger
+    var
+        DraftOrderId: BigInteger;
+        GraphQuery: TextBuilder;
+    begin
+        GraphQuery := CreateDraftOrderGQLRequest(TempShpfyOrderHeader, TempShpfyOrderLine, ShpfyOrderTaxLines);
+        DraftOrderId := SendDraftOrderGraphQLRequest(GraphQuery);
+        exit(DraftOrderId);
+    end;
+
+    /// <summary>
+    /// Completes a draft order in shopify by converting it to an order.
+    /// </summary>
+    /// <param name="DraftOrderId">Draft order id that needs to be completed.</param>
+    /// <returns>Json response of a created order in shopify.</returns>
+    internal procedure CompleteDraftOrder(DraftOrderId: BigInteger): JsonToken
+    var
+        GraphQLType: Enum "Shpfy GraphQL Type";
+        Parameters: Dictionary of [Text, Text];
+        JResponse: JsonToken;
+    begin
+        GraphQLType := "Shpfy GraphQL Type"::DraftOrderComplete;
+        Parameters.Add('DraftOrderId', Format(DraftOrderId));
+        JResponse := ShpfyCommunicationMgt.ExecuteGraphQL(GraphQLType, Parameters);
+        exit(JResponse);
+    end;
+
+    /// <summary>
+    /// Sets a global shopify shop to be used for draft orders api functionality.
+    /// </summary>
+    /// <param name="ShopCode">Shopify shop code to be set.</param>
+    internal procedure SetShop(ShopCode: Code[20])
+    begin
+        Clear(ShpfyShop);
+        ShpfyShop.Get(ShopCode);
+        ShpfyCommunicationMgt.SetShop(ShpfyShop);
+    end;
+
+    local procedure CreateDraftOrderGQLRequest(
+        var TempShpfyOrderHeader: Record "Shpfy Order Header" temporary;
+        var TempShpfyOrderLine: Record "Shpfy Order Line" temporary;
+        var ShpfyOrderTaxLines: Dictionary of [Text, Decimal]
+    ): TextBuilder
+    var
+        GraphQuery: TextBuilder;
+    begin
+        GraphQuery.Append('{"query":"mutation {draftOrderCreate(input: {');
+        if TempShpfyOrderHeader.Email <> '' then begin
+            GraphQuery.Append('email: \"');
+            GraphQuery.Append(ShpfyCommunicationMgt.EscapeGrapQLData(TempShpfyOrderHeader.Email));
+            GraphQuery.Append('\"');
+        end;
+        if TempShpfyOrderHeader."Phone No." <> '' then begin
+            GraphQuery.Append('phone: \"');
+            GraphQuery.Append(ShpfyCommunicationMgt.EscapeGrapQLData(TempShpfyOrderHeader."Phone No."));
+            GraphQuery.Append('\"');
+        end;
+        if TempShpfyOrderHeader."Currency Code" <> '' then begin
+            GraphQuery.Append('presentmentCurrencyCode: ');
```

Comment:
A comma is missing here between `phone `and `presentmentCurrencyCode`, so an order with different currency than LCY would fail to create in Shopify. This leads me to believe that this code path is never tested. That's why I asked for unit tests check for coverage but dismissed as "they were not part of the scope for this particular task".

I would think any development task should include unit tests by default in today's world. And bugs like these would have been discovered if unit tested properly. This PR is already merged so there's no action to take but this is something we will definitely keep in mind for future projects.